File: comparison/results/haiku/codecraft/humaneval/HumanEval_154/javascript/solution.js
function cycpattern_check(a, b){
  for (let i = 0; i < b.length; i++) {
    const rotation = b.substring(i) + b.substring(0, i);
    if (a.includes(rotation)) {
      return true;
    }
  }
  return false;
}

File: comparison/results/haiku/codecraft/humaneval/HumanEval_154/java/solution.java
public static boolean cycpatternCheck(String a, String b) {
    for (int i = 0; i < b.length(); i++) {
        String rotated = b.substring(i) + b.substring(0, i);
        if (a.contains(rotated)) {
            return true;
        }
    }
    return false;
}
